src/librustc_mir/interpret/place.rs

```diff
@@ -216,21 +216,40 @@ impl<'tcx, Tag> MPlaceTy<'tcx, Tag> {
             _ => bug!("vtable not supported on type {:?}", self.layout.ty),
         }
     }
+
+    #[inline(always)]
+    pub fn to_const_value(self, tcx: TyCtxt<'tcx>) -> ConstValue<'tcx> {
```

Comment:
This is an odd function... why is it here, and not in `const_eval.rs` where we have most of the other conversion functions?

Reply:
Also, converting a *place* to a *value* doesn't make much sense to begin with. The returned value represents the content of the place, right (the "pointee")?

Reply:
true. I could name this function better.

I'll also move it.

Reply:
it's kind of a "read". But since a `Place` is just an "rvalue reference" in c++ terms, converting it to a value seemed ok to me.

I inlined the function as a closure at the use site